src/app/guards/auth-guard.service.ts

```diff
@@ -0,0 +1,21 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { CanActivate, Router, UrlTree } from '@angular/router';
+import { AuthService } from '../services/auth.service';
+
+@Injectable()
+export class AuthGuard implements CanActivate {
+  constructor(private authService: AuthService, private router: Router) {}
+
+  canActivate(): Observable<boolean> | Promise<boolean> | boolean | UrlTree {
+    const routePath = '/auth/login';
+    const isAuthenticated = this.authService.isLoggedIn();
+
+    if (isAuthenticated) {
+      return true;
+    }
+
+    const tree: UrlTree = this.router.parseUrl(routePath);
```

Comment:
I am not sure how this will help. I think `UrlTree` was introduced as return type from Angular 7 version.

Isn't it we can do this as well simply:

`return isAuthenticated`

?

Reply:
We can't because it will behave the same way as above

Reply:
Check out the GIF in the PR description. You will see the difference